components/hubspot/hubspot.app.mjs

```diff
@@ -172,138 +145,285 @@ export default {
             value,
           };
         });
-        return options;
       },
     },
   },
   methods: {
-    _client() {
-      return new hubspotSDK.Client({
-        accessToken: this.$auth.oauth_access_token,
-      });
+    _getBaseURL() {
+      return "https://api.hubapi.com";
     },
-    monthAgo() {
-      const monthAgo = new Date();
-      monthAgo.setMonth(monthAgo.getMonth() - 1);
-      return monthAgo;
+    _getHeaders() {
+      return {
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+        "Content-Type": "application/json",
+      };
     },
-    async makeAxiosRequest(method, endpoint, params = null) {
+    async makeRequest(endpoint, opts = {}) {
+      const {
+        method = "GET",
+        params,
+        data,
+        $,
+      } = opts;
       const config = {
         method,
-        url: `https://api.hubapi.com${endpoint}`,
-        headers: {
-          "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
-          "Content-Type": "application/json",
-        },
+        url: `${this._getBaseURL()}${endpoint}`,
+        headers: this._getHeaders(),
         params,
+        data,
       };
-      return (await axios(config)).data;
+      return axios($ ?? this, config);
     },
-    async makeRequest(method, path, body = null) {
-      const client = this._client();
-      return client.apiRequest({
-        method,
-        path,
-        body,
+    async makeGetRequest(endpoint, params = null, $) {
+      return this.makeRequest(endpoint, {
+        params,
+        $,
+      });
+    },
+    async makePostRequest({
+      endpoint, data,
+    }, $) {
+      return this.makeRequest(endpoint, {
+        method: "POST",
+        data,
+        $,
       });
     },
+    /**
+     * Returns the object type name of a CRM object to be used in Hubspot API
+     * calls
+     *
+     * {@see {@link https://developers.hubspot.com/docs/cms/data/crm-objects CRM Object}
+     *
+     * @param {string} objectSlug - The object type's plural label or fully qualified
+     * name
+     * @returns The object type name
+     */
+    getObjectTypeName(objectSlug) {
+      if (!objectSlug.endsWith("s")) {
+        return objectSlug.toLowerCase();
+      }
+      if (objectSlug === "companies") {
+        return "company";
+      }
+      return objectSlug.toLowerCase().slice(0, -1);
+    },
+    /**
+     * Returns a label for a CRM object, intended to be used as the label for
+     * prop options
+     *
+     * @param {object} object - The CRM object to get a label for
+     * @param {string} objectSlug - The object type's plural label or fully
+     * qualified name
+     * @returns A label for the object
+     */
+    getObjectLabel(object, objectSlug) {
+      const {
+        id,
+        properties,
+      } = object;
+      const objectName = this.getObjectTypeName(objectSlug);
+      switch (objectName) {
```

Comment:
I think this is not necessary since you are controlling the slug from UI with a given array of options right? something like this I think would do it as well
```suggestion
      switch (objectSlug) {
```